src/commands/utils/graph.ts

```diff
@@ -0,0 +1,231 @@
+        return { succeeded: true, result: application };
+    } catch (e) {
+        return { succeeded: false, error: getErrorMessage(e) };
+    }
+}
+
+export async function deleteApplication(graphClient: Client, applicationObjectId: string): Promise<Errorable<void>> {
+    try {
+        await graphClient.api(`/applications/${applicationObjectId}`).delete();
+        return { succeeded: true, result: undefined };
+    } catch (e) {
+        return { succeeded: false, error: getErrorMessage(e) };
+    }
+}
+
+export async function createServicePrincipal(
+    graphClient: Client,
+    applicationId: string,
+): Promise<Errorable<ServicePrincipal>> {
+    const newServicePrincipal: ServicePrincipalParams = {
+        appId: applicationId,
+    };
+
+    try {
+        const servicePrincipal: ServicePrincipal = await graphClient
+            .api("/servicePrincipals")
+            .post(newServicePrincipal);
+        return { succeeded: true, result: servicePrincipal };
+    } catch (e) {
+        return { succeeded: false, error: getErrorMessage(e) };
+    }
+}
+
+export async function getServicePrincipalsForApp(
+    graphClient: Client,
+    appId: string,
+): Promise<Errorable<ServicePrincipal[]>> {
+    try {
+        const spSearchResults: GraphListResult<ServicePrincipal> = await graphClient
+            .api("/servicePrincipals")
+            .filter(`appId eq '${appId}'`)
+            .get();
+
+        return { succeeded: true, result: spSearchResults.value };
+    } catch (e) {
+        return { succeeded: false, error: getErrorMessage(e) };
+    }
+}
+
+export async function getFederatedIdentityCredentials(
+    graphClient: Client,
+    applicationId: string,
+): Promise<Errorable<FederatedIdentityCredential[]>> {
+    try {
+        const identityResults: GraphListResult<FederatedIdentityCredential> = await graphClient
+            .api(`/applications/${applicationId}/federatedIdentityCredentials`)
+            .get();
+
+        return { succeeded: true, result: identityResults.value };
+    } catch (e) {
+        return { succeeded: false, error: getErrorMessage(e) };
+    }
+}
+
+export async function createFederatedIdentityCredential(
+    graphClient: Client,
+    applicationId: string,
+    subject: string,
+    name: string,
+    description: string,
+): Promise<Errorable<FederatedIdentityCredential>> {
+    const newCred: FederatedIdentityCredentialParams = {
+        name,
+        subject,
+        issuer: federatedIdentityCredentialIssuer,
+        description,
+        audiences: [federatedIdentityCredentialAudience],
+    };
+
+    try {
+        const cred: FederatedIdentityCredential = await graphClient
+            .api(`/applications/${applicationId}/federatedIdentityCredentials`)
+            .post(newCred);
+
+        return { succeeded: true, result: cred };
+    } catch (e) {
+        return { succeeded: false, error: getErrorMessage(e) };
+    }
+} ////////TODO: catch fail logic if the same cred already exists
```

Comment:
You want the TODO's to handle in this PR or next set of PR's? As far as I have tested createFedCreds, the error message contains credentials already exists error which you can show it as warning or notification in the UX.